drivers/bmp180/bmp180.c

```diff
@@ -240,9 +240,9 @@ static int _read_up(const bmp180_t *dev, int32_t *output)
 
 static int _compute_b5(const bmp180_t *dev, int32_t ut, int32_t *output)
 {
-    int32_t x1, x2;
-    x1 = (ut - dev->calibration.ac6) * dev->calibration.ac5 >> 15;
-    x2 = (dev->calibration.mc << 11) / (x1 + dev->calibration.md);
+    int32_t x1 = 0, x2 = 0;
+    x1 = ((int32_t)ut - (int32_t)dev->calibration.ac6) * (int32_t)dev->calibration.ac5 >> 15;
```

Comment:
Adding the type casts is OK. But furthermore there is a set of parenthesis missing. Take a look at the manufacturer's driver.

    x1 = (((int32_t)ut - (int32_t)dev->calibration.ac6) * (int32_t)dev->calibration.ac5) >> 15;